src/network-uptime/network-dashboard.py

```diff
@@ -27,11 +29,17 @@
 selected_device = st.sidebar.selectbox('Device name', device_name)
 
 
-@st.cache
+@st.cache(suppress_st_warning=True)
 def load_device_data(device):
-    api = f"{BASE_API_URL}/monitor/devices/uptime?tenant=airqo&startDate=2022-01-01T00:00:00.001Z&endDate=2022-03-25T19:00:00.001Z&device_name=' + device
+    api = f"{BASE_API_URL}/monitor/devices/uptime?tenant=airqo&startDate={START_QUERY_DATE}&endDate={END_QUERY_DATE}&device_name=" + device
```

Comment:
@uman95, just curious why the start time and end time are environment variables. Won't this limit a user’s ability to customize the date ranges or require updating the .env whenever we want to increase the date range. What about using an interval. some thing like 

```python
interval_in_days = os.getenv('INTERVAL_IN_DAYS', 30)
end_date = datetime.utcnow()
start_date = end_date - timedelta(days=interval_in_days)
START_QUERY_DATE = datetime.strftime(start_date, "%Y-%m-%dT00:00:00Z")
END_QUERY_DATE = datetime.strftime(end_date, "%Y-%m-%dT11:59:59Z")
```
Here we can adjust how the date
Incase you are using months, then you can utilise the function below to get the first and last day of a month

```python
def first_day_of_month(date_time: datetime) -> datetime:
    return datetime(year=date_time.year, month=date_time.month, day=1)

def last_day_of_month(date_time: datetime) -> datetime:
    month = date_time.month
    if month < 12:
        month = month + 1
        next_month = datetime(year=date_time.year, month=month, day=1)
        return next_month - timedelta(days=1)
    else:
        year = date_time.year + 1
        next_month = datetime(year=year, month=1, day=1)
        return next_month - timedelta(days=1)
```

Reply:
@NoahNsimbe This PR mainly fix errors from newly added device. The date part is a short term fix (what we currently have actually is the Q1 network uptime, reason for the fixed date range). The whole thing is a work in progress, considering suggestions we got from the last team updates.

Reply:
cool @uman95